test/libsolidity/semanticTests/expressions/tuple_from_ternary_expression.sol

```diff
@@ -0,0 +1,8 @@
+contract C {
+    function f(bool flag) public pure returns (uint, uint, uint){
+        return (flag ? (1, 2, 3) : (3, 2, 1));
```

Comment:
So what about side-effects here? In https://github.com/ethereum/solidity/pull/14371#discussion_r1259895931 I meant that we should check whether the `flag = true` bit actually flips the flag on legacy. This test does not verify that.

Actually, I *really* hope this case is not broken, because, as this example shows there are sensible uses for ternary with tuples. I think disallowing it would be a proper breaking change. So if it's broken, we'll probably have to handle it the same way as #14366, i.e. by actually evaluating it in the codegen (in a separate PR of course).

Reply:
Now that I think about it, there's a high chance that this is not affected - because, unlike in the other cases, the result cannot be inferred from the AST without evaluating the expression. For example there's no way to know the result of this without evaluation:

```
return (flag ? (x, y, z) : (a, b, c));
```

So we're probably safe, but we should still cover it with test cases.

Reply:
> So what about side-effects here? In [#14371 (comment)](https://github.com/ethereum/solidity/pull/14371#discussion_r1259895931) I meant that we should check whether the `flag = true` bit actually flips the flag on legacy. This test does not verify that.

Right, I missed the point there.

> So we're probably safe, but we should still verify and cover it with test cases.

Updated the test and fortunately it is not breaking.